### ISSUE TITLE
Translate int.Parse and similar

### ISSUE THREAD

**Issue by roji** [dotnet/efcore#28287]
We translate `Convert<int>(string)`, but we could also translate x.Parse on the various primitive types (int, long...), as they're a common .NET API. We'd only support the simple overloads accepting a string, nothing more.

Originally raised in https://github.com/dotnet/efcore/issues/28282#issuecomment-1161782149

**Comment by roji** [comment]
Yeah, hopefully One Day(TM) we'll have infra for identifying code inside EF Core expression trees, and then we'd be able to suppress those warnings. This is similar to various string methods, null-ref warnings #21663, etc. - added a note under #22086.

**Comment by MaRK0960** [comment]
I'm interested in working on this issue.

I'm planning on creating an SqlServerParseTranslator similar to SqlServerConvertTranslator.

https://github.com/dotnet/efcore/blob/781ca2451d0d408813acebe87f2fcaa03d4b62e2/src/EFCore.SqlServer/Query/Internal/SqlServerConvertTranslator.cs

But the `IEnumerable<MethodInfo> SupportedMethods` will be the x.Parse methods with one overload.

Regarding the tests cases, those are the ones I managed to find:

https://github.com/dotnet/efcore/blob/781ca2451d0d408813acebe87f2fcaa03d4b62e2/test/EFCore.SqlServer.FunctionalTests/Query/NorthwindFunctionsQuerySqlServerTest.cs#L1285-L1619

The `Convert_ToBoolean` doesn't test against `Convert.ToBoolean(Convert.ToString(x))` translating to `CONVERT(bit, CONVERT(nvarchar(max), x))`

Is there a reason for that?

**Comment by roji** [comment]
@MaRK0960 the issue is that string representations of booleans has lots of options (true/false/TRUE/FALSE/1/0?), and these aren't the same across C# and the database. I think it makes sense to leave out bool.TryParse specifically here.
